cmd/influxd/handler.go

```diff
@@ -29,6 +30,21 @@ func NewHandler() *Handler {
 
 // ServeHTTP responds to HTTP request to the handler.
 func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	// Debug routes.
+	if strings.HasPrefix(r.URL.Path, "/debug/pprof") {
+		switch r.URL.Path {
+		case "/debug/pprof/cmdline":
+			pprof.Cmdline(w, r)
+		case "/debug/pprof/profile":
+			pprof.Profile(w, r)
+		case "/debug/pprof/symbol":
+			pprof.Symbol(w, r)
+		default:
+			pprof.Index(w, r)
+		}
+		return
+	}
+
```

Comment:
You can now pull pprof profiles if you run:

```sh
$ go tool pprof http://localhost:8086/debug/pprof/profile
```

More info on the [`net/http/pprof`](https://golang.org/pkg/net/http/pprof/) page. Make sure you change their `:6060` ports to `:8086` though.